src/pybind/mgr/dashboard_v2/controllers/cephfs.py

```diff
@@ -0,0 +1,321 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import collections
+from collections import defaultdict
+import json
+import time
+
+import cherrypy
+from mgr_module import CommandResult
+
+from ..tools import ApiController, AuthRequired, BaseController, NotificationQueue, ViewCache
+
+
+LOG_BUFFER_SIZE = 30
+
+
+@ApiController('cephfs')
+@AuthRequired()
+class CephFS(BaseController):
+    def __init__(self):
+        super(CephFS, self).__init__()
+
+        # Stateful instances of CephFSClients, hold cached results.  Key to
+        # dict is FSCID
+        self.cephfs_clients = {}
+
+    @cherrypy.expose
+    @cherrypy.tools.json_out()
+    def clients(self, fs_id):
+        fs_id = self.fs_id_to_int(fs_id)
+
+        return self._clients(fs_id)
+
+    @cherrypy.expose
+    @cherrypy.tools.json_out()
+    def data(self, fs_id):
+        fs_id = self.fs_id_to_int(fs_id)
+
+        return self.fs_status(fs_id)
+
+    @cherrypy.expose
+    @cherrypy.tools.json_out()
+    def mds_counters(self, fs_id):
+        """
+        Result format: map of daemon name to map of counter to list of datapoints
+        rtype: dict[str, dict[str, list]]
+        """
+
+        # Opinionated list of interesting performance counters for the GUI --
+        # if you need something else just add it.  See how simple life is
+        # when you don't have to write general purpose APIs?
+        counters = [
+            "mds_server.handle_client_request",
+            "mds_log.ev",
+            "mds_cache.num_strays",
+            "mds.exported",
+            "mds.exported_inodes",
+            "mds.imported",
+            "mds.imported_inodes",
+            "mds.inodes",
+            "mds.caps",
+            "mds.subtrees"
+        ]
+
+        fs_id = self.fs_id_to_int(fs_id)
+
+        result = {}
+        mds_names = self._get_mds_names(fs_id)
+
+        for mds_name in mds_names:
+            result[mds_name] = {}
+            for counter in counters:
+                data = self.mgr.get_counter("mds", mds_name, counter)
+                if data is not None:
+                    result[mds_name][counter] = data[counter]
+                else:
+                    result[mds_name][counter] = []
+
+        return dict(result)
+
+    def fs_id_to_int(self, fs_id):
```

Comment:
As you've added this to the class, it can be marked as `@staticmethod`

Reply:
done